sourcegraph/cody
# Fix duplication of query when selecting a context file that already exists earlier in the input

The code here appeared to try to preserve suffixes after the filename, but we appear to only support selecting files at the end of the input anyway. By assuming the index of "fileDisplayText" in the input was the one that the user expected to replace, an input like:

```
@foo.txt @fo
```

Would assume the first `@foo.txt` is being replaced, and retain `@fo` (actually duplicating it).

Removing this suffix fixes the issue and as far as I can tell does not break any other functionality. The issue described in https://github.com/sourcegraph/cody/issues/1973 still appears to work as expected with this change.

Fixes #2243:

https://github.com/sourcegraph/cody/assets/1078012/53055f8b-9b61-49e4-bdf7-66afa4367599

#1973 is still fixed:

https://github.com/sourcegraph/cody/assets/1078012/9053b3ce-0b27-494a-8243-7276406cc6d3


@abeatrix Please take a look, because this partially reverts your changes from #1980. If you think there are cases that this is breaking, let me know and I will add tests and try to fix another way.

## Test plan

- Run `pnpm test:e2e at-file` to run the updated test
- Open chat and try to mention a file a second time without typing the fill name (as shown in the first video above)
- Open chat, complete a filename, then backspace over the added trailing space and replace it with a `?` and ensure it's still used as a context file (second video above)
